### ISSUE TITLE
Profile set in spring.profiles.active is ignored if the same profile is set as spring.profiles.default as well

### ISSUE THREAD

**Issue by deepu105** [spring-projects/spring-boot#6833]
This is related to issue https://github.com/spring-projects/spring-boot/issues/5998 Im opening a new issue as per suggestion by @philwebb 
I was trying out the fix from #5998 in version 1.4 and it is working for the most part. There is one small glitch though consider the below

1. I have two profiles `dev` and `prod`
2. I have set `spring.profiles.default` to `dev` using 
```
public static void addDefaultProfile(SpringApplication app) {
        Map<String, Object> defProperties =  new HashMap<>();
        defProperties.put("spring.profiles.default", "dev");
        app.setDefaultProperties(defProperties);
    }
```
Note: I'm not using `Collections.singletonMap` as its makes the array fixed size, which causes issues as you would try to add stuff to that array in spring
3. Now if I do `env.acceptsProfiles("dev")` it will return false, which is OK since I didn't set it as active profile
4. now if I do `--spring.profiles.active=dev` I would expect `env.acceptsProfiles("dev")` to return true but it doesn't. seems like if the same profile is used as active and default the default one is used. Ideally it should be promoted to active

The use-case for this is if I want to do some stuff based on profiles I might be using `env.acceptsProfiles("dev")` to check if the profile is active and in this case its failing.

Here is a test case for the scenario, Ideally the case should pass, but with the latest spring boot release(1.4) version its failing
```
@Test
public void sameDefaultProfileAndActive() throws Exception {
 		SpringApplication application = new SpringApplication(Config.class);
 		application.setWebEnvironment(false);
 		this.context = application.run("--spring.profiles.default=customdefault",
 				"--spring.profiles.active=customdefault");
 		assertThat(this.context.getEnvironment().acceptsProfiles("customdefault"))
 				.isTrue();
}
```

**Comment by philwebb** [comment]
@deepu105 The test example you've provided seems to pass for me. Can you provide a GitHub project that shows the failure?

**Comment by deepu105** [comment]
Ok thats weird. Ill share a sample project asap

**Comment by deepu105** [comment]
@philwebb sorry about that I guess I gave the wrong test case above, its passing in my setup as well. Here  is the case which is failing and by further digging I found out that in normal scenario this works, but this fails when there is a profile included in any loaded file.

consider the below test case this **passes**
```
	@Test
	public void customDefaultProfileAndActiveFromFile() throws Exception {
		// gh-5998
		SpringApplication application = new SpringApplication(Config.class);
		application.setWebEnvironment(false);
		this.context = application.run("--spring.config.name=application",
				"--spring.profiles.default=customdefault");
        assertThat(this.context.getEnvironment().acceptsProfiles("customdefault"))
            .isTrue();
	}
```
the config file loaded is `application.yml`
```
spring:
    profiles:
        active: customdefault
```

the customdefault config file `application-customdefault.yml` is
```
my.property: fromdevpropertiesfile
```

Now for the same test case if you replace the `application-customdefault.yml` with below  it **fails**

```
spring:
    profiles:
        include: some-profile

my.property: fromdevpropertiesfile
```

Shouldnt this pass as well?
Note: It fails only when active profile is set from a file, if its set via `"--spring.profiles.active=customdefault"` it passes
Hope this helps

**Comment by philwebb** [comment]
I might not be totally getting this, perhaps you can update the sample to show exactly what you mean?

**Comment by deepu105** [comment]
@philwebb Ok to put it simply the below test case should pass with the property files below
```
	@Test
	public void customDefaultProfileAndActiveFromFile() throws Exception {
		// gh-5998
		SpringApplication application = new SpringApplication(Config.class);
		application.setWebEnvironment(false);
		this.context = application.run("--spring.config.name=application",
				"--spring.profiles.default=customdefault");
        assertThat(this.context.getEnvironment().acceptsProfiles("customdefault"))
            .isTrue();
	}
```
the config file loaded is `application.yml`
```
spring:
    profiles:
        active: customdefault
```

the customdefault profile config file `application-customdefault.yml` is
```
spring:
    profiles:
        include: some-profile

my.property: fromdevpropertiesfile
```

**Comment by philwebb** [comment]
Thanks for bearing with me! This should now be fixed in the latest 1.4.1 SNAPSHOT.

**Comment by deepu105** [comment]
Thank you

Thanks & regards,
Deepu
